### PR TITLE
feat(server): migrate pascal case http headers rule to context object to support openapi 3

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Related to #714

### CODE REVIEW

File: server/src/test/java/de/zalando/zally/rule/zalando/PascalCaseHttpHeadersRuleTest.kt [hunk 1]
```diff
@@ -1,68 +1,45 @@
...
 
-    private val rule = PascalCaseHttpHeadersRule(testConfig)
+    private val rule = PascalCaseHttpHeadersRule()
 
     @Test
-    fun simplePositiveCase() {
-        val swagger = swaggerWithHeaderParams("Right-Name")
-        assertThat(rule.validate(swagger)).isNull()
-    }
+    fun `checkHttpHeaders should return violation if not pascal case is used`() {
```
Comment (tkrop) [inline server/src/test/java/de/zalando/zally/rule/zalando/PascalCaseHttpHeadersRuleTest.kt]: This one negative check is definitely not enough as long as we do not have tests for `de.zalando.zally.util.OpenApiUtil.kt`. We need to add a check for each potential header location, to make sure that we do not miss one. Anyhow, I would prefer similar unit tests on `de.zalando.zally.util.OpenApiUtil.kt`.

Comment (maxim-tschumak) [reply server/src/test/java/de/zalando/zally/rule/zalando/PascalCaseHttpHeadersRuleTest.kt]: Agree, we should unit test `OpenApiUtil` class.

**Comment by maxim-tschumak** [comment]
@tkrop please re-review, the improvements have been implemented

**Comment by tkrop** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
@tkrop please re-review, the improvements have been implemented

**Pull request merged**